### PR TITLE
fix: install necessary packages for node-gyp, use multi-stage build

### PR DESCRIPTION
**Pull request by thebengeu**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,10 +1,21 @@
-FROM mhart/alpine-node:14
+FROM mhart/alpine-node:16
```
Comment (darora) [inline Dockerfile]: Consider using the official `node:16-alpine3` images

Comment (thebengeu) [reply Dockerfile]: Done.

Comment (inian) [reply Dockerfile]: lets upgrade to 16 in a separate PR? I really want to upgrade to 16 too (native M1 node builds 🎉 ), but our [CI](https://github.com/supabase/storage-api/blob/master/.github/workflows/ci.yml) and [build](https://github.com/supabase/storage-api/blob/be6d3f5b1923594bda1dbad6ea21eae97ea51dad/.github/workflows/build.yml#L49) needs to be updated too. If not, our self-hosted version will be based on node16 and prod will be in node14 and there might be a mismatch there.

Comment (thebengeu) [reply Dockerfile]: Switched back to 14.

**Review by inian — APPROVED** [review]

**Comment by inian** [comment]
Good catch Beng! `fs-xattr` is a native module which requires node-gyp

**Pull request merged**